src/vmm/src/system_allocator.rs

```diff
@@ -0,0 +1,28 @@
+use crate::{MemoryConfig, MemoryError};
+use vm_allocator::{AddressAllocator, AllocPolicy, IdAllocator, RangeInclusive};
+use vm_device::bus::{BusRange, MmioAddress, MmioRange};
+pub(crate) struct SystemAllocator {
+    address: AddressAllocator,
+}
+pub type Result<T> = std::result::Result<T, crate::Error>;
+
+impl SystemAllocator {
```

Comment:
Why are you implementing a wrapper over the `AddressAllocator`? Do you think we will be using more than one allocator? If we don't use another one, it might be a good option to just use `AddressAllocator` directly so that we don't need the wrapper methods.

Reply:
This is an early optimization, we don't know how the code will look like at that point in time, so I think it would be better instead to just use the AddressAllocator directly.

Otherwise, the integration is going in the good direction, can you do the same for the serial console, and other devices?